aws/resource_aws_cloudtrail.go

```diff
@@ -72,6 +73,46 @@ func resourceAwsCloudTrail() *schema.Resource {
 				Optional:     true,
 				ValidateFunc: validateArn,
 			},
+			"event_selector": &schema.Schema{
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 5,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"read_write_type": &schema.Schema{
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validation.StringInSlice([]string{"All", "ReadOnly", "WriteOnly"}, false),
+						},
+
+						"include_management_events": &schema.Schema{
+							Type:     schema.TypeBool,
+							Optional: true,
+							Default:  true,
+						},
+
+						"data_resource": &schema.Schema{
+							Type:     schema.TypeSet,
```

Comment:
This might be easier to work with as a `schema.TypeList` (especially for acceptance testing) unless there's a good reason to keep it a set.